src/ArmTemplates/Extractor/EntityExtractors/ParametersExtractor.cs

```diff
@@ -119,6 +120,39 @@ async Task AddServiceUrlParameterAsync()
                 parameters.Add(ParameterNames.ServiceUrl, new TemplateObjectParameterProperties() { Value = serviceUrls });
             }
 
+            async Task AddApiOauth2ScopeParameterAsync()
+            {
+                if (!extractorParameters.ParametrizeApiOauth2Scope)
+                {
+                    return;
+                }
+
+                var apiOauth2Scopes = new Dictionary<string, string>();
+                foreach (var apiName in apisToExtract)
+                {
+                    var apiDetails = await this.apisClient.GetSingleAsync(apiName, extractorParameters);
```

Comment:
we can make a single request like `this.apisClient.GetAllAsync()` and then iterate through all apis instead of iterating and calling `GetSingleAsync` every time